python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -212,6 +230,41 @@ def check_for_partitions(cls, values: dict):
             raise ValueError("Must provide partitions if using parallel")
         return values
 
+    @root_validator(pre=True)
+    def validate_hydrofabic(cls, values: dict) -> dict:
+        """
+
+        Args:
+            values (dict): configuation values to check
+
+        Raises:
+            ValueError: If a geopackage hydrofabric or set of geojsons are not found
+
+        Returns:
+            dict: Valid configuration elements for hydrofabric requirements
+        """
+        hf: FilePath = values.get('hydrofabric')
+        cats: FilePath = values.get("catchments")
+        nex: FilePath = values.get("nexus")
+        x: FilePath = values.get("crosswalk")
+
+        if hf is None and cats is None and nex is None and x is None:
+            msg = "Must provide a geopackage input with the hydrofabric key"\
+                  "or proide catchment, nexus, and crosswalk geojson files."  
+            raise ValueError(msg)
+        
+        if(hf is not None):
+            try:
+                p = Path(hf)
+                if not p.exists():
+                    raise
+            except:
+                raise TypeError("hydrofabric must be a valid file path")
+        if cats is not None or nex is not None or x is not None:
+            print("WARNING: GeoJSON support will be deprecated in a future release, use geopackage hydrofabric.")
```

Comment:
Lets use `warnings.warn` here instead.

Reply:
Yeah, that...I was trying to think of it at the time, and drew a huge blank 🤣

Reply:
😆!

Reply:
changed in [4db27d0](https://github.com/NOAA-OWP/ngen-cal/pull/128/commits/4db27d0b2a9b3fd66adad0db26f1079b0e3dd14a)